ee/management/commands/migrate_clickhouse.py

```diff
@@ -71,18 +79,35 @@ def migrate(self, host, options):
                 )
             print("Migrations done")
         else:
+            is_fresh_instance = len(self.get_applied_migrations(database)) == 0
             database.migrate(MIGRATIONS_PACKAGE_NAME, options["upto"], replicated=CLICKHOUSE_REPLICATION)
-            print("Migration successful")
+            if is_fresh_instance and not options["skip_special_migrations"]:
+                for migration_name, definition in ALL_SPECIAL_MIGRATIONS.items():
+                    sm = SpecialMigration.objects.get_or_create(
+                        name=migration_name,
+                        description=definition.description,
+                        posthog_min_version=definition.posthog_min_version,
+                        posthog_max_version=definition.posthog_max_version,
+                    )[0]
+                    if sm.status == MigrationStatus.NotStarted:
+                        print("Applying special migration", migration_name)
+                        started_successfully = start_special_migration(migration_name)
+                        sm.refresh_from_db()
+                        if not started_successfully or sm.status != MigrationStatus.CompletedSuccessfully:
+                            print(f"Unable to complete special migration {migration_name} with error", sm.last_error)
+                            return
+            print("✅ Migration successful")
 
     def get_migrations(self, database, upto):
-        applied_migrations = database._get_applied_migrations(
-            MIGRATIONS_PACKAGE_NAME, replicated=CLICKHOUSE_REPLICATION
-        )
         modules = import_submodules(MIGRATIONS_PACKAGE_NAME)
+        applied_migrations = self.get_applied_migrations(database)
         unapplied_migrations = set(modules.keys()) - applied_migrations
 
         for migration_name in sorted(unapplied_migrations):
```

Comment:
If we use sort I think we should make sure we **enforce** a prefix in all our migrations. Otherwise we might end up applying them in the wrong order

Reply:
actually this is not for special migrations - this was here already and is for clickhouse migrations